napari/util/misc.py

```diff
@@ -227,21 +227,58 @@ def slice_image(data, indices, multichannel=False):
         Indices to slice array at.
     multichannel : bool, optional
         Flag if image is multichannel, for example RGB or RGBA.
+    return_indices : bool, optional
+        Flag if returning sliced image data or a slice tuple.
```

Comment:
I prefer using a different function here. It's generally considered bad practice to change the return type based on input values. It's also unclear from the title why you would use a function to put in indices and get out indices, so probably what you want is a helper function, `clip_indices`, that gets called by `slice_image`?

Reply:
i've dropped this function, wasn't really needed